pandas/tseries/tools.py

```diff
@@ -409,22 +429,47 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
             except (ValueError, TypeError):
                 raise e
 
+    # perform the conversion
+    if origin == 'julian':
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        try:
+            arg = arg - tslib.Timestamp(0).to_julian_date()
+        except:
+            raise ValueError("incompatible 'arg' type for given "
+                             "'origin'='julian'")
     if arg is None:
-        return arg
+        result = arg
     elif isinstance(arg, tslib.Timestamp):
-        return arg
+        result = arg
     elif isinstance(arg, ABCSeries):
         from pandas import Series
         values = _convert_listlike(arg._values, False, format)
-        return Series(values, index=arg.index, name=arg.name)
+        result = Series(values, index=arg.index, name=arg.name)
     elif isinstance(arg, (ABCDataFrame, MutableMapping)):
-        return _assemble_from_unit_mappings(arg, errors=errors)
+        result = _assemble_from_unit_mappings(arg, errors=errors)
     elif isinstance(arg, ABCIndexClass):
-        return _convert_listlike(arg, box, format, name=arg.name)
+        result = _convert_listlike(arg, box, format, name=arg.name)
     elif is_list_like(arg):
-        return _convert_listlike(arg, box, format)
+        result = _convert_listlike(arg, box, format)
+    else:
+        result = _convert_listlike(np.array([arg]), box, format)[0]
 
-    return _convert_listlike(np.array([arg]), box, format)[0]
+    # handle origin
+    if origin not in ['unix', 'julian']:
+        try:
+            offset = tslib.Timestamp(origin) - tslib.Timestamp(0)
+        except tslib.OutOfBoundsDatetime:
+            raise ValueError(
+                "origin {} is Out of Bounds".format(origin))
```

Comment:
No, I mean for origin specifically (by circumventing the `Timestamp` creation), to have something like `pd.to_datetime(.., origin='0001-01-01')` working if the end timestamp is not out of bounds.

Eg by using `dateutil.parser.parse(origin)`, but then we still need to convert that to the correct offset